src/OptFrame/Heuristics/EvolutionaryAlgorithms/PSO.hpp

```diff
@@ -264,6 +113,22 @@ class PSO : public SingleObjSearch<XES, XES2, XSH2>
         }
         return swarm;
     }
+    
+    void BirdMutation ( Bird b)
```

Comment:
This Bird is being passed by copy... so, it's useless to try to change it.
You need to pass this by reference: Bird& b